docs/platforms/python/integrations/celery/index.mdx

```diff
@@ -17,36 +17,89 @@ pip install --upgrade 'sentry-sdk[celery]'
+Make sure that the **call to `sentry_sdk.init()` is loaded on worker startup**, and not only in the module where your tasks are defined. Otherwise, the initialization happens too late and events might end up not being reported.
 
-<PlatformContent includePath="getting-started-config" />
+### Setup Celery (Without Django)
 
-### Standalone Setup
+To get the most out of Sentry make sure to initialize the Sentry SDK in your Celery worker processes as well as your application that is sending messages to Celery.
 
-If you're using Celery standalone, there are two ways to set this up:
+In addition to capturing errors, you can monitor interactions between multiple services or applications by [enabling tracing](/concepts/key-terms/tracing/). You can also collect and analyze performance profiles from real users with [profiling](/product/explore/profiling/).
 
-- Initializing the SDK in the configuration file loaded with Celery's `--config` parameter
-- Initializing the SDK by hooking it to either the [`celeryd_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#celeryd-init) or [`worker_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#worker-init) signals
+Select which Sentry features you'd like to install in addition to Error Monitoring to get the corresponding installation and configuration instructions below.
 
-  ```python
-  import sentry_sdk
-  from celery import Celery, signals
+#### Setup in Celery
 
-  app = Celery("myapp")
+<OnboardingOptionButtons
+  options={["error-monitoring", "performance", "profiling"]}
+/>
 
-  #@signals.worker_init.connect
-  @signals.celeryd_init.connect
-  def init_sentry(**_kwargs):
-      sentry_sdk.init(...)  # same as above
-  ```
+```python {filename:tasks.py} {"onboardingOptions": {"performance": "12-14", "profiling": "15-18"}}
+from celery import Celery, signals
+import sentry_sdk
+
+# Initializing Celery
+app = Celery("tasks", broker="...")
+
+# Initialize Sentry SDK on Celery startup
+@signals.celeryd_init.connect
+def init_sentry(**_kwargs):
+    sentry_sdk.init(
+        dsn="___PUBLIC_DSN___",
+        # Set traces_sample_rate to 1.0 to capture 100%
+        # of transactions for tracing.
+        traces_sample_rate=1.0,
+        # Set profiles_sample_rate to 1.0 to profile 100%
+        # of sampled transactions.
+        # We recommend adjusting this value in production.
+        profiles_sample_rate=1.0,
+    )
+
+# Task definitions go here
+@app.task
+def add(x, y):
+    return x + y
+```
+
+The [`celeryd_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#celeryd-init) signal is triggered when the Celery deamon is started, before the worker processes are spawned. You can use the [`worker_init`](https://docs.celeryq.dev/en/stable/userguide/signals.html?#worker-init) signal instead if you want to initialize Sentry on start of each worker process.
+
+#### Setup in Your Application
```

Comment:
I find it confusing as well....

Reply:
Celery is confusing at first. 

The application and Celery stuff are completely separate. Those are two entirely separate processes you need to start: Your application and on the other hand the Celery. daemon This can also be on different machines. They are only connected through the broker (a message queue like RabbitMQ or Redis) So you have to have two Sentry.init() calls, one in your application sending messages to Celery and one in Celery processing those messages. 

People who use Celery know this and are probably not that confused. 

(Django is a special case, because you have the the Sentry config in Django and this config is used by the Django process and by the Celery process) Celery was kind of built for Django and then established itself in other setups too. 

I added some better headlines and a bit more text to reduce confusion.